tests/Feature/Router/RouterRunTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GacelaTest\Feature\Router;
+
+use Gacela\Router\Entities\Request;
+use Gacela\Router\Entities\Response;
+use Gacela\Router\Router;
+use Gacela\Router\Routes;
+use GacelaTest\Feature\HeaderTestCase;
+
+final class RouterRunTest extends HeaderTestCase
+{
+    public function test_run_multiple_route_collections(): void
+    {
+        $_SERVER['REQUEST_URI'] = 'https://example.org/uri-2';
+        $_SERVER['REQUEST_METHOD'] = Request::METHOD_GET;
+
+        $router = Router::create();
+
+        $router->addRoutes(static function (Routes $routes): void {
+            $routes->get('uri-1', static fn () => new Response('first body'));
+        });
+
+        $router->addRoutes(static function (Routes $routes): void {
+            $routes->get('uri-2', static fn () => new Response('second body'));
+        });
+
+        $router->run();
+
+        $this->expectOutputString('second body');
+
+        self::assertSame([], $this->headers());
+    }
+
+    public function test_run_same_multiple_route_collections_then_first_with_priority(): void
```

Comment:
Why does the first one have priority?
I would say even the opposite, the latter overrides if that route existed before 🤔

Reply:
When iterating over the route collection, the first one to be found is the winner. The main motivation for this is that you shouldn't have the same path pointing to 2 different actions. Anyway, let's consider the pros vs const:
**PROS**: it's "inheritance oriented", the route define latest win the race. The question is: Do you want to even have a "race" in this terms?
**CONST**: You need to iterate over all possible routes all the time, looking if there is defined a duplicate path anywhere. This delay the performance for no reason in my opinion.
What do you think, @Tito-Kati?

Reply:
If you register two routes with the same URI it is your (the developer's) problem.

It's not something we have to manage unless you want to throw an exception.

Even so, this is somewhat complex because there are routes that can collide due to parameters, etc...

They are also resource-consuming checks to make sure that you (the developer) have configured your routes correctly.

I advocate not only not doing anything to fix it but also not refer to it in any test, if it is not in our specifications (our tests) it is not a feature, therefore we have no obligation to support it.

Reply:
TL;DR: I think the best thing is to make this test like the one above but checking that the second route works and I wouldn't worry about anything else.